### PR TITLE
fixed scripts w/solr8 ckan core

### PR DESCRIPTION
**Pull request by azimhamza**
**Pull Request: Update CKAN Core & SOLR8 Integration Fixes**

### What this PR accomplishes:
1. **CKAN Core & SOLR8 Integration**:
    - Addressed critical issues regarding CKAN core not updating in tandem with changes pushed to SOLR8.
    - Steps accomplished:
        1. Stopping the solr8 server.
        2. Removing all temporary files.
        3. Creating new configuration directories.
        4. Copying configuration files, including `managed-schema`, `solrconfig.xml`, `protwords.txt`, `stopwords.txt`, and `synonyms.txt`, from the ckan ontario-theme to the appropriate directories.
        5. Restarting the Solr8 server.
        6. Creating CKAN core.
    - Additionally, specific fixes have been applied to the `synonyms.txt` and `protwords.txt` files.

2. **Testing & Replacement of Files**:
    - Instructions have been provided to ensure that any changes made can be tested and that the default files can be replaced with the required ones.

### What needs review:
1. **CKAN Core & SOLR8 Integration Steps**:
    - Review the process of stopping, configuring, and restarting the Solr8 server.
    - Validate the file paths and commands for updating the CKAN core configuration, including the newly fixed stopwords.txt`synonyms.txt` and `protwords.txt`.

**Additional Notes**:
This PR ensures that the proper files have been loaded and integrates the fixed configurations. The script has been corrected, and detailed instructions have been included for manual replacement and testing of files. It's vital to stop the CKAN server, apply the required changes, and then restart the CKAN core for testing.

Please review the changes and confirm the integration process is aligned with the project's requirements.

### CODE REVIEW

File: scripts/setup_solr.sh [hunk 1]
```diff
@@ -38,7 +38,19 @@ cp $REPODIR/../config/solr/managed-schema .
 echo $SUDOPASS | sudo -S -k chmod 775 managed-schema
 echo "Schema fetched."
 
-echo "Creating a ckan core..."
-echo $SUDOPASS | sudo -S -k -u solr /opt/solr/bin/solr create -c ckan
-echo "ckan core created."
+# New CKAN core creation method
+echo "Updating CKAN core with new method..."
+sudo -u solr /opt/solr/bin/solr stop -all
```
Comment (aimalkhan) [inline scripts/setup_solr.sh:43]: should be `echo $SUDOPASS | sudo -S -k` format rather than `sudo`

File: scripts/setup_solr.sh [hunk 2]
```diff
@@ -38,7 +38,19 @@ cp $REPODIR/../config/solr/managed-schema .
 echo $SUDOPASS | sudo -S -k chmod 775 managed-schema
 echo "Schema fetched."
 
-echo "Creating a ckan core..."
-echo $SUDOPASS | sudo -S -k -u solr /opt/solr/bin/solr create -c ckan
-echo "ckan core created."
+# New CKAN core creation method
+echo "Updating CKAN core with new method..."
```
Comment (aimalkhan) [inline scripts/setup_solr.sh:42]: this msg does not make sense

File: scripts/setup_solr.sh [hunk 3]
```diff
@@ -38,7 +38,19 @@ cp $REPODIR/../config/solr/managed-schema .
...
+sudo rm -rf /var/solr/data/ckan
+sudo mkdir -p /opt/solr/server/solr/configsets/ckan_conf/conf
+sudo cp /usr/lib/ckan/default/src/ckanext-ontario_theme/config/solr/managed-schema /opt/solr/server/solr/configsets/ckan_conf/conf
+sudo cp /lib/ckan/default/src/ckanext-ontario_theme/config/solr/solrconfig.xml /opt/solr/server/solr/configsets/ckan_conf/conf
+sudo cp /lib/ckan/default/src/ckanext-ontario_theme/config/solr/protwords.txt /opt/solr/server/solr/configsets/ckan_conf/conf
+sudo cp /lib/ckan/default/src/ckanext-ontario_theme/config/solr/stopwords.txt /opt/solr/server/solr/configsets/ckan_conf/conf
+sudo cp /lib/ckan/default/src/ckanext-ontario_theme/config/solr/synonyms.txt /opt/solr/server/solr/configsets/ckan_conf/conf
+sudo -u solr /opt/solr/bin/solr restart
+sudo -u solr /opt/solr/bin/solr create -c ckan -d /opt/solr/server/solr/configsets/ckan_conf/conf
+echo "CKAN core updated with new method."
```
Comment (aimalkhan) [inline scripts/setup_solr.sh:53]: this msg does not make sense

**Review by aimalkhan — CHANGES_REQUESTED** [review]
the PR message does not make sense in context of the changes and the script needs changes.

File: scripts/setup_solr.sh (hunk 2 above)
Comment (azimhamza) [reply scripts/setup_solr.sh:42]: fixed

File: scripts/setup_solr.sh (hunk 3 above)
Comment (azimhamza) [reply scripts/setup_solr.sh:53]: fixes to "ckan core created""

File: scripts/setup_solr.sh (hunk 1 above)
Comment (azimhamza) [reply scripts/setup_solr.sh]: changed

**Review by azimhamza — COMMENTED** [review]
Fixed the Given PR issues given by Aimal.

**Review by aimalkhan — APPROVED** [review]
installation script now works

**Pull request merged**